coin_purse.gd

```diff
@@ -0,0 +1,24 @@
+extends Resource
+class_name CoinPurse
+
+var _coins: int = 0
+
+func _init() -> void:
+	SignalBus.enemy_died.connect(_handle_enemy_died)
+
+func get_coins() -> int:
+	return _coins
+
+func spend_coins(amount: int) -> void:
+	_transact(-amount)
+
+func earn_coins(amount: int) -> void:
+	_transact(amount)
+
+func _transact(amount: int) -> void:
+	_coins = max(_coins + amount, 0)
+	SignalBus.coins_changed.emit(_coins)
```

Comment:
this brings it down to on place for emitting coins_changed which is nice.

i wonder how things are done without a global `SignalBus` - not that we need to change anything, was just wondering about that

Reply:
If we don't have it, then whatever needs to listen to the changes will also need a reference to the object itself.

Probably being the HUD and the Upgrade Menu, which, honestly either they already have a purse, or they should get one.

The SignalBus works great, however, I think the less we use it, the "better" the design is.

SignalBus kind of blurries the dependencies, but it's great to have for certain things.